File: app/src/main/java/com/samsalek/foodbankandroid/retrofit/DishApi.java
package com.samsalek.foodbankandroid.retrofit;

import com.samsalek.foodbankandroid.model.Dish;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;

public interface DishApi {

    String baseUrl = "/api/v1/dish";

    @GET(baseUrl + "/all")
    Call<List<Dish>> getDishes();

    @POST(baseUrl + "/add")
    Call<Dish> addDish(@Body Dish dish);
}

File: app/src/main/java/io/github/samsalmag/foodbankandroid/retrofit/DishApi.kt
package io.github.samsalmag.foodbankandroid.retrofit

import io.github.samsalmag.foodbankandroid.model.Dish
import retrofit2.http.GET

import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST

interface DishApi {

    companion object {
        const val baseUrl = "/api/v1/dish"
    }

    @GET(baseUrl + "/all")
    fun dishes(): Call<List<Dish>>

    @POST(baseUrl + "/add")
    fun addDish(@Body dish: Dish): Call<Dish>
}
